Snowflake.Data/Core/FastParser.cs

```diff
@@ -100,7 +101,7 @@ public static decimal FastParseDecimal(byte[] s, int offset, int len)
                 catch (OverflowException)
                 {
                     // Fallback to regular decimal constructor from string instead.
-                    return decimal.Parse(UTF8Buffer.UTF8.GetString(s, offset, len));
+                    return decimal.Parse(UTF8Buffer.UTF8.GetString(s, offset, len), CultureInfo.InvariantCulture);
```

Comment:
Before making code changes, please create a doc for short coming of existing code and what we are going to do to support Locale. We need to find out all areas missing Locale support.